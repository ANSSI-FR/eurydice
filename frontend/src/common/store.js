import Vue from "vue";
import Vuex from "vuex";

Vue.use(Vuex);

export default new Vuex.Store({
  state: () => ({
    username: null,
    maintenance: false,
    serverDown: false,
    serverMetadata: null,
    ongoingTransfer: false,
  }),
  getters: {
    maintenance(state) {
      return state.maintenance;
    },
    serverDown(state) {
      return state.serverDown;
    },
    serverMetadata(state) {
      return state.serverMetadata;
    },
    ongoingTransfer(state) {
      return state.ongoingTransfer;
    },
  },
  mutations: {
    setUsername(state, newUsername) {
      state.username = newUsername;
    },
    setMaintenance(state, newMaintenance) {
      state.maintenance = newMaintenance;
    },
    setServerDown(state, serverDown) {
      state.serverDown = serverDown;
    },
    setServerMetadata(state, serverMetadata) {
      state.serverMetadata = serverMetadata;
    },
    setOngoingTransfer(state, ongoingTransfer) {
      state.ongoingTransfer = ongoingTransfer;
    },
  },
  actions: {},
  // NOTE: consider adding modules to this common store in main.js
  //       when adding origin/destination specific state/mutations
  modules: {},
});
