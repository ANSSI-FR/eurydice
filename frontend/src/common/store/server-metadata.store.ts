import * as serverMetadataService from '@common/services/server-metadata.service';
import { defineStore } from 'pinia';
import { computed, ref } from 'vue';

export const useServerMetadataStore = defineStore('serverMetadataStore', () => {
  const serverMetadata = ref<ServerMetadata>();

  const getServerMetadata = computed(() => serverMetadata.value);
  const setServerMetadata = (metadata: ServerMetadata | undefined) => {
    serverMetadata.value = metadata;
  };
  const fetchServerMetadata = async () => {
    await serverMetadataService.getServerMetadata().then((fetchedServerMetadata) => {
      setServerMetadata(fetchedServerMetadata);
    });
  };
  const $reset = () => {
    serverMetadata.value = undefined;
  };
  return {
    getServerMetadata,
    setServerMetadata,
    fetchServerMetadata,
    $reset,
  };
});
