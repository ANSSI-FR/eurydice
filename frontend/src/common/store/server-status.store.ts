import { defineStore } from 'pinia';
import { computed, ref } from 'vue';

export const useServerStatusStore = defineStore('serverStatusStore', () => {
  const isServerDown = ref<boolean>(false);
  const isServerInMaintenance = ref<boolean>(false);

  const getIsServerDown = computed(() => isServerDown.value);
  const setIsServerDown = (value: boolean) => {
    isServerDown.value = value;
  };

  const getIsServerInMaintenance = computed(() => isServerInMaintenance.value);
  const setIsServerInMaintenance = (value: boolean) => {
    isServerInMaintenance.value = value;
  };

  const $reset = () => {
    isServerDown.value = false;
    isServerInMaintenance.value = false;
  };
  return {
    getIsServerDown,
    setIsServerDown,
    getIsServerInMaintenance,
    setIsServerInMaintenance,
    $reset,
  };
});
