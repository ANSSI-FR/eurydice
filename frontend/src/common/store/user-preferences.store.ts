import type { UserPreferences } from '@common/models/UserPreferences';
import { defineStore } from 'pinia';
import { readonly, ref } from 'vue';

export const useUserPreferencesStore = defineStore('userPreferencesStore', () => {
  const LOCAL_STORAGE_KEY: string = 'userPreferences';

  const updateDarkModeInHTML = (darkMode: boolean | undefined) => {
    if (darkMode) {
      document.querySelector('html')!.classList.add('dark');
    } else {
      document.querySelector('html')!.classList.remove('dark');
    }
  };

  const initFromLocalStorage = () => {
    let preferences;
    const localStoragePreferences = localStorage.getItem(LOCAL_STORAGE_KEY);
    if (localStoragePreferences) {
      preferences = JSON.parse(localStoragePreferences);
    } else {
      const systemDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches;
      preferences = { darkMode: systemDarkMode };
    }
    updateDarkModeInHTML(preferences.darkMode);
    return preferences;
  };

  const saveUserPreferencesToLocalStorage = () => {
    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(userPreferences.value));
  };

  const userPreferences = ref<UserPreferences | undefined>(initFromLocalStorage());

  const getUserPreferences = readonly(userPreferences);

  const setDarkModePreference = (darkMode: boolean) => {
    if (!userPreferences.value) {
      userPreferences.value = {};
    }
    if (userPreferences.value.darkMode !== darkMode) {
      userPreferences.value.darkMode = darkMode;
      saveUserPreferencesToLocalStorage();
      updateDarkModeInHTML(darkMode);
    }
  };

  return {
    getUserPreferences,
    setDarkModePreference,
  };
});
