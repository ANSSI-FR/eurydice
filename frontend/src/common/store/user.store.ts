import { defineStore } from 'pinia';
import { computed, ref } from 'vue';

export const useUserStore = defineStore('userStore', () => {
  const currentUser = ref<User>();

  const getCurrentUser = computed(() => currentUser.value);
  const setCurrentUser = (user: User | undefined) => {
    currentUser.value = user;
  };
  const $reset = () => {
    currentUser.value = undefined;
  };
  return {
    getCurrentUser,
    setCurrentUser,
    $reset,
  };
});
