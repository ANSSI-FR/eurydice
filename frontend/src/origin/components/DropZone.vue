<template>
  <div
    @drop.prevent="uploadFilesFromEvent"
    @dragover.prevent
    @dragenter.prevent
  >
    <v-tabs bg-color="primary">
      <v-tab @click="changeTab('files')">Fichiers</v-tab>
      <v-tab @click="changeTab('text')">Texte</v-tab>
    </v-tabs>
    <v-file-input
      v-if="formatTab == 'files'"
      v-model="files"
      :clearable="false"
      chips
      counter
      multiple
      outlined
      show-size
      prepend-icon
      :webkitdirectory="allowDirectoryUpload"
      :allowdirs="allowDirectoryUpload"
      :disabled="disableDropZone"
      :placeholder="
        disableDropZone
          ? 'Le transfert de fichiers n\'est pas opérationnel'
          : 'Cliquez ou déposez vos fichiers ici'
      "
      @change="uploadFiles"
    >
      <template v-if="files.length === 0" #prepend-inner>
        <v-icon v-if="disableDropZone" large>{{ mdiTrafficCone }}</v-icon>
        <v-icon v-else large>{{ mdiUpload }}</v-icon>
      </template>
      <template #selection="{ file }">
        <DropZoneFile
          :name="file.name"
          :progress="file.progress"
          :finished="file.state === STATES.SUCCESS"
          :error="file.state === STATES.FAILURE"
        ></DropZoneFile>
      </template>
      <template v-if="files.length !== 0" #append>
        <div class="DropZone-icons">
          <v-tooltip left>
            <template #activator="{ on, attrs }">
              <v-icon v-bind="attrs" v-on="on" @click="removeFinished">{{
                mdiCheckAll
              }}</v-icon>
            </template>
            <span>Masquer les transferts terminés</span>
          </v-tooltip>
        </div>
      </template>
    </v-file-input>
    <TextTransferZone v-if="formatTab == 'text'" @send_text="uploadFile" />
  </div>
</template>

<script>
import {
  mdiUpload,
  mdiCheckAll,
  mdiDeleteSweep,
  mdiTrafficCone,
} from "@mdi/js";
import { mapGetters } from "vuex";
import DropZoneFile from "@origin/components/DropZoneFile";
import {
  createTransferable,
  validateTransferable,
} from "@origin/api/transferables";
import axios from "axios";
import TextTransferZone from "@origin/components/TextTransferZone";

const STATES = {
  ONGOING: "ongoing",
  SUCCESS: "success",
  FAILURE: "failure",
  CANCELED: "canceled",
};

export default {
  name: "DropZone",
  components: { DropZoneFile, TextTransferZone },
  data: () => {
    return {
      files: [],
      oldFiles: [],
      STATES,
      allowDirectoryUpload: false,
      fileUploadPromises: [],
      mdiUpload,
      mdiCheckAll,
      mdiDeleteSweep,
      mdiTrafficCone,
      formatTab: "files",
    };
  },
  computed: {
    ...mapGetters(["serverDown", "maintenance"]),
    disableDropZone() {
      return !this.maintenance && this.serverDown;
    },
  },
  methods: {
    beforeUnloadListener(event) {
      event.preventDefault();
      /* eslint-disable-next-line no-param-reassign */ /* eslint-disable-next-line no-return-assign */
      return (event.returnValue =
        "Attention, les téléversements en cours depuis cet onglet vers le guichet origine seront annulés.");
    },
    uploadFilesFromEvent(event) {
      const items = Array.from(event.dataTransfer.items);

      // anticipate potential rename of webkitGetAsEntry to getAsEntry
      // https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/webkitGetAsEntry
      const getAsEntry = (item) =>
        item.webkitGetAsEntry?.() || item.getAsEntry?.();

      const newFiles = items
        .filter((item) => getAsEntry(item)?.isFile)
        .map((file) => file.getAsFile());

      this.uploadFiles(newFiles);

      if (items.some((item) => getAsEntry(item)?.isDirectory)) {
        this.$alert(
          "Il n'est pas possible d'envoyer des dossiers, ceux-ci ont été ignorés.",
          "warning"
        );
      }
    },
    uploadFiles(newFiles) {
      window.addEventListener("beforeunload", this.beforeUnloadListener, {
        capture: true,
      });

      if (newFiles === this.oldFiles) return;

      this.files = [...this.oldFiles, ...newFiles];
      this.oldFiles = this.files;

      this.fileUploadPromises.push(
        Promise.all(newFiles.map((file) => this.uploadFile(file)))
      );
    },
    async uploadFile(fileReference) {
      const file = fileReference;
      this.annotateFile(file);

      try {
        validateTransferable(file);
        const data = await createTransferable(file);
        file.state =
          data && data.submissionSucceeded ? STATES.SUCCESS : STATES.FAILURE;
      } catch (error) {
        if (error?.response) {
          file.state = STATES.FAILURE;
          // Request was throttled or the file's size exceeds the limit
          if ([413, 429].includes(error?.response.status)) {
            this.$alert(error.response.data.detail, "error");
          }
        } else if (error instanceof axios.Cancel) {
          file.state = STATES.CANCELED;
          this.removeOne(this.files.indexOf(file));
          this.$alert("Le transfert a été interrompu", "info");
        }
      } finally {
        file.progress = 100;
        this.refreshFile(file);

        Promise.all(this.fileUploadPromises).then(() => {
          window.removeEventListener(
            "beforeunload",
            this.beforeUnloadListener,
            {
              capture: true,
            }
          );
        });
      }
    },
    annotateFile(fileReference) {
      const file = fileReference;
      file.progress = 0;
      file.state = STATES.ONGOING;
      file.abortController = new AbortController();
      file.progressUpdater = this.generateFileProgressUpdater(file);
      this.refreshFile(file);
    },
    generateFileProgressUpdater(fileReference) {
      const file = fileReference;
      return (progress) => {
        file.progress = Math.round((progress.loaded * 100) / progress.total);
        this.refreshFile(file);
      };
    },
    refreshFile(file) {
      // Required for Vue to notice a change to `this.files`
      const index = this.files.indexOf(file);
      if (index !== -1) this.$set(this.files, index, file);
    },
    removeOne(index = 0) {
      const removed = this.files.splice(index, 1)[0];
      this.oldFiles = this.files;

      if (removed.state === STATES.ONGOING) removed.abortController.abort();
    },
    removeAll() {
      while (this.files.length) this.removeOne();
    },
    removeFinished() {
      this.files = this.files.filter((f) => f.state === STATES.ONGOING);
      this.oldFiles = this.files;
    },
    changeTab(currentTab) {
      this.formatTab = currentTab;
    },
  },
};
</script>

<style lang="scss" scoped>
::v-deep {
  .v-input__append-inner {
    align-self: center;
    margin-top: 0;
  }

  .v-input__prepend-inner {
    left: calc(50% - 18px);
    position: absolute;
    top: 12px;
  }

  .v-file-input__text--placeholder {
    align-content: center !important;
    font-size: 1.2em;
    justify-content: center;
    padding-top: 40px;
  }

  .v-file-input__text {
    align-content: flex-start;
    align-items: flex-start;
    min-height: 120px;
  }

  .DropZone-icons {
    display: flex;
    flex-flow: column nowrap;
    gap: 16px;
  }
}
</style>
