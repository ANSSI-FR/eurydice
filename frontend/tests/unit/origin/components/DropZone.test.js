import {
  createTransferable,
  validateTransferable,
} from "@origin/api/transferables";
import DropZone from "@origin/components/DropZone";
import { createLocalVue, mount } from "@vue/test-utils";
import Vue from "vue";
import store from "@/common/store";
import Vuetify from "vuetify";
import { Cancel } from "axios";

jest.mock("@origin/api/transferables", () => ({
  createTransferable: jest.fn(),
  validateTransferable: jest.fn(),
}));

describe("DropZone", () => {
  let wrapper;

  const mockedFile =
    "A society that devours its own young deserves no automatic or unquestioning allegiance";
  const fileEntry = {
    isFile: true,
  };
  const directoryEntry = {
    isFile: false,
    isDirectory: true,
  };
  const fileDataTransferItem = {
    webkitGetAsEntry: () => fileEntry,
    getAsFile: () => mockedFile,
  };
  const fileDataTransferItemWithGetAsEntry = {
    getAsEntry: () => fileEntry,
    getAsFile: () => mockedFile,
  };
  const directoryDataTransferItem = {
    webkitGetAsEntry: () => directoryEntry,
  };

  beforeEach(() => {
    const localVue = createLocalVue();
    const vuetify = new Vuetify();
    wrapper = mount(DropZone, { localVue, vuetify, store });
  });
  afterEach(() => {
    jest.resetAllMocks();
  });
  it("renders correctly when sender is working", async () => {
    store.commit("setServerDown", false);
    store.commit("setMaintenance", false);
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders correctly when sender is not working", async () => {
    store.commit("setServerDown", true);
    store.commit("setMaintenance", false);
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot();
  });
  it("updates files to ONGOING and SUCCESS states", async () => {
    let createTransferableReturn = null;
    createTransferable.mockReturnValue(
      new Promise((resolve) => {
        createTransferableReturn = resolve;
      })
    );

    wrapper.vm.uploadFiles([{ name: "coucou.exe" }]);
    await wrapper.vm.$nextTick();

    expect(createTransferable).toHaveBeenCalled();
    expect(wrapper.vm.files[0].state).toBe(wrapper.vm.STATES.ONGOING);
    expect(wrapper.find("span.v-chip").classes()).not.toContain("blue");

    createTransferableReturn({ submissionSucceeded: true });
    await wrapper.vm.$nextTick();
    await wrapper.vm.$nextTick();

    expect(wrapper.vm.files[0].state).toBe(wrapper.vm.STATES.SUCCESS);
    expect(wrapper.find("span.v-chip").classes()).toContain("blue");
  });
  it("updates progress as the file is uploaded", async () => {
    const file = { name: "coucou.exe" };
    // Mock API call so that it doesn't return immediately.
    // Else, the progress would immediately be 100%.
    createTransferable.mockReturnValue(new Promise(() => {}));
    wrapper.vm.uploadFiles([file]);
    await wrapper.vm.$nextTick();

    expect(wrapper.find(".v-progress-linear").attributes("aria-valuenow")).toBe(
      "0"
    );

    file.progressUpdater({ loaded: 84, total: 200 });
    await wrapper.vm.$nextTick();

    expect(wrapper.find(".v-progress-linear").attributes("aria-valuenow")).toBe(
      "42"
    );
  });
  it("shows 100% progress on empty file upload", async () => {
    const file = { name: "coucou.exe", size: 0 };
    createTransferable.mockReturnValue({
      id: "00002800-0000-1000-8000-00805f9b34fb",
    });

    wrapper.vm.uploadFiles([file]);
    await wrapper.vm.$nextTick();

    expect(wrapper.find(".v-progress-linear").attributes("aria-valuenow")).toBe(
      "100"
    );
  });
  it("calls uploadFilesFromEvent on drop event", async () => {
    wrapper.vm.uploadFilesFromEvent = jest.fn();
    wrapper.find(".drop-zone").trigger("drop");

    expect(wrapper.vm.uploadFilesFromEvent).toHaveBeenCalledTimes(1);
  });
  it("uploads files in drop event", async () => {
    const event = {
      dataTransfer: {
        items: [fileDataTransferItem],
      },
    };
    wrapper.vm.uploadFiles = jest.fn();

    wrapper.vm.uploadFilesFromEvent(event);

    expect(wrapper.vm.uploadFiles).toHaveBeenCalledTimes(1);
    expect(wrapper.vm.uploadFiles).toHaveBeenCalledWith([mockedFile]);
  });
  it("only uploads files in drop event", async () => {
    const event = {
      dataTransfer: {
        items: [fileDataTransferItem, directoryDataTransferItem],
      },
    };
    wrapper.vm.uploadFiles = jest.fn();

    wrapper.vm.uploadFilesFromEvent(event);

    expect(wrapper.vm.uploadFiles).toHaveBeenCalledTimes(1);
    expect(wrapper.vm.uploadFiles).toHaveBeenCalledWith([mockedFile]);
  });
  it("alerts when directories are in drop event", async () => {
    Vue.prototype.$alert = jest.fn();
    const event = {
      dataTransfer: {
        items: [directoryDataTransferItem],
      },
    };
    wrapper.vm.uploadFiles = jest.fn();

    wrapper.vm.uploadFilesFromEvent(event);

    expect(Vue.prototype.$alert).toHaveBeenCalledWith(
      "Il n'est pas possible d'envoyer des dossiers, ceux-ci ont été ignorés.",
      "warning"
    );
  });
  it("has a file input field allowing multiple files but forbidding directory upload", async () => {
    const nativeFileInput = wrapper.get("input[type='file']");

    expect(nativeFileInput.attributes()).toHaveProperty("multiple");
    expect(nativeFileInput.attributes()).not.toHaveProperty("webkitdirectory");
    expect(nativeFileInput.attributes()).not.toHaveProperty("allowdirs");
  });
  it("uploads files in drop event if webkitGetAsEntry has been replaced by getAsEntry", async () => {
    const event = {
      dataTransfer: {
        items: [fileDataTransferItemWithGetAsEntry],
      },
    };
    wrapper.vm.uploadFiles = jest.fn();

    wrapper.vm.uploadFilesFromEvent(event);

    expect(wrapper.vm.uploadFiles).toHaveBeenCalledTimes(1);
    expect(wrapper.vm.uploadFiles).toHaveBeenCalledWith([mockedFile]);
  });
  it("alerts when http429 is received because backend is throttling", async () => {
    Vue.prototype.$alert = jest.fn();

    createTransferable.mockReturnValue(
      new Promise((_, reject) => {
        const error = new Error("HTTP 429");
        error.response = {
          status: 429,
          data: { detail: "Request was throttled." },
        };
        reject(error);
      })
    );

    wrapper.vm.uploadFiles([{ name: "coucou.exe" }]);
    await wrapper.vm.$nextTick();

    expect(Vue.prototype.$alert).toHaveBeenCalledWith(
      "Request was throttled.",
      "error"
    );
  });
  it("alerts when http413 is received because the file is too large", async () => {
    Vue.prototype.$alert = jest.fn();

    validateTransferable.mockImplementation(() => {
      const error = new Error("Content Too Large.");
      error.response = {
        status: 413,
        data: { detail: "Content Too Large." },
      };
      throw error;
    });

    wrapper.vm.uploadFiles([{ name: "dump.big" }]);
    await wrapper.vm.$nextTick();

    expect(Vue.prototype.$alert).toHaveBeenCalledWith(
      "Content Too Large.",
      "error"
    );
  });
  it("handles transfer cancellations", async () => {
    Vue.prototype.$alert = jest.fn();

    // The "canceled" string is hardcoded in the axios' source code:
    // https://github.com/axios/axios/blob/1025d1231a7747503188459dd5a6d1effdcea928/lib/adapters/xhr.js#L194
    createTransferable.mockRejectedValue(new Cancel("canceled"));

    wrapper.vm.uploadFiles([{ name: "foo.bar" }]);

    const file = wrapper.vm.files[0];

    await wrapper.vm.$nextTick();
    await wrapper.vm.$nextTick();

    expect(file.state).toBe(wrapper.vm.STATES.CANCELED);
    expect(Vue.prototype.$alert).toHaveBeenCalledWith(
      "Le transfert a été interrompu",
      "info"
    );
  });
});
