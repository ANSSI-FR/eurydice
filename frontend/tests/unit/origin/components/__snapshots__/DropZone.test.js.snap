// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropZone renders correctly when sender is not working 1`] = `
<div>
  <div
    bg-color="primary"
    class="v-tabs theme--light"
  >
    <div
      class="v-item-group theme--light v-slide-group v-tabs-bar v-tabs-bar--is-mobile primary--text"
      role="tablist"
    >
      <div
        class="v-slide-group__prev v-slide-group__prev--disabled"
      >
        <transition-stub
          name="fade-transition"
        />
      </div>
      <div
        class="v-slide-group__wrapper"
      >
        <div
          class="v-slide-group__content v-tabs-bar__content"
        >
          <div
            class="v-tabs-slider-wrapper"
            style="transition: none;"
          >
            <div
              class="v-tabs-slider"
            />
          </div>
          <div
            aria-selected="true"
            class="v-tab v-tab--active"
            role="tab"
            tabindex="0"
          >
            Fichiers
          </div>

          <div
            aria-selected="false"
            class="v-tab"
            role="tab"
            tabindex="0"
          >
            Texte
          </div>
        </div>
      </div>
      <div
        class="v-slide-group__next v-slide-group__next--disabled"
      >
        <transition-stub
          name="fade-transition"
        />
      </div>
    </div>
  </div>

  <div
    class="drop-zone"
  >
    <div
      class="v-input v-input--is-disabled theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder v-file-input"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-input__prepend-inner"
          >
            <span
              aria-hidden="true"
              class="v-icon notranslate theme--light"
              style="font-size: 36px; height: 36px; width: 36px;"
            >
              <svg
                aria-hidden="true"
                class="v-icon__svg"
                role="img"
                style="font-size: 36px; height: 36px; width: 36px;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 15L18 19H21V22H3V19H6L7 15H17M15 8L16 12H8L9 8H15M13 1L14 5H10L11 1H13Z"
                />
              </svg>
            </span>
          </div>
          <fieldset
            aria-hidden="true"
          >
            <legend
              style="width: 0px;"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
          <div
            class="v-text-field__slot"
          >
            <div
              class="v-file-input__text v-file-input__text--placeholder"
            >
              Le transfert de fichiers n'est pas opérationnel
            </div>
            <input
              disabled="disabled"
              id="input-24"
              multiple="multiple"
              placeholder="Le transfert de fichiers n'est pas opérationnel"
              type="file"
            />
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
          <div
            class="v-counter theme--light"
          >
            0 files (0 B in total)
          </div>
        </div>
      </div>
    </div>
  </div>

  <!---->
</div>
`;

exports[`DropZone renders correctly when sender is working 1`] = `
<div>
  <div
    bg-color="primary"
    class="v-tabs theme--light"
  >
    <div
      class="v-item-group theme--light v-slide-group v-tabs-bar v-tabs-bar--is-mobile primary--text"
      role="tablist"
    >
      <div
        class="v-slide-group__prev v-slide-group__prev--disabled"
      >
        <transition-stub
          name="fade-transition"
        />
      </div>
      <div
        class="v-slide-group__wrapper"
      >
        <div
          class="v-slide-group__content v-tabs-bar__content"
        >
          <div
            class="v-tabs-slider-wrapper"
            style="transition: none;"
          >
            <div
              class="v-tabs-slider"
            />
          </div>
          <div
            aria-selected="true"
            class="v-tab v-tab--active"
            role="tab"
            tabindex="0"
          >
            Fichiers
          </div>

          <div
            aria-selected="false"
            class="v-tab"
            role="tab"
            tabindex="0"
          >
            Texte
          </div>
        </div>
      </div>
      <div
        class="v-slide-group__next v-slide-group__next--disabled"
      >
        <transition-stub
          name="fade-transition"
        />
      </div>
    </div>
  </div>

  <div
    class="drop-zone"
  >
    <div
      class="v-input theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-text-field--placeholder v-file-input"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-input__prepend-inner"
          >
            <span
              aria-hidden="true"
              class="v-icon notranslate theme--light"
              style="font-size: 36px; height: 36px; width: 36px;"
            >
              <svg
                aria-hidden="true"
                class="v-icon__svg"
                role="img"
                style="font-size: 36px; height: 36px; width: 36px;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9,16V10H5L12,3L19,10H15V16H9M5,20V18H19V20H5Z"
                />
              </svg>
            </span>
          </div>
          <fieldset
            aria-hidden="true"
          >
            <legend
              style="width: 0px;"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
          <div
            class="v-text-field__slot"
          >
            <div
              class="v-file-input__text v-file-input__text--placeholder"
            >
              Cliquez ou déposez vos fichiers ici
            </div>
            <input
              id="input-11"
              multiple="multiple"
              placeholder="Cliquez ou déposez vos fichiers ici"
              type="file"
            />
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
          <div
            class="v-counter theme--light"
          >
            0 files (0 B in total)
          </div>
        </div>
      </div>
    </div>
  </div>

  <!---->
</div>
`;
